### DISCUSSION TITLE
Tutorial or settings

### DISCUSSION THREAD

**Discussion by vitos1k** [bezo97/IFSRenderer#35] [Q&A] [answered]
Hi there! Very great software! It's magnificient! 
I've got a question, there is a very beautifull image  in the header of the website https://ifsrenderer.z97.io/
![background](https://github.com/bezo97/IFSRenderer/assets/1842518/11b89af3-5d9c-4413-97b4-4b0a98cd7970)

Is there a tutorial or guide on how to create this image or sort of images?

**Comment by bezo97** [comment, marked as answer]
Thanks for the kind words. This is an old randomly generated fractal I did during development back in 2020. Sadly those params are not compatible with the current version, but I did my best to convert it: [test8.zip](https://github.com/bezo97/IFSRenderer/files/15418820/test8.zip). I was not able to convert the affine transform's Scale and Rotation, but its a good start to play around. Also you'll need to install the "Foci" plugin included in the zip (you can just drag&drop it to the window. Feel free to post your results in the forum!

**Reply by vitos1k** [reply]
Thank you very much! this is awesome, will tinker around!

**Comment by MaxStarovoyt** [comment]
IFSRenderer is the Softwere from aliens world). Thank you for create! And question: how I can change colours in gradients?

**Reply by bezo97** [reply]
Glad to hear :D For now you can only "replace" the gradient by loading from a file or generating a new random one (in the generator). These gradient pack files can be found mostly on deviantart, search for "apophysis gradient pack" or so.
But I got great news for you, recently I've started working on a gradient editor, which will let the user modify existing gradients and manage their favorite color palettes. Here's a sneak peek of the current state 👍 
![image](https://github.com/user-attachments/assets/c4b77b69-4452-481a-a27a-ffa483775449)
Probably coming in the next release!

**Reply by MaxStarovoyt** [reply]
Cool! Thank you and good luck!

**Comment by MaxStarovoyt** [comment]
Hello! Could you give some recommendations to optimize rendering speed/quality. I've already seen excellent denoising performance (it's amazing) and mostly use animation target level 15. But maybe there are some other ways to optimize results? something like density cutter,  for example, trim off what is not very visible in the render. I will be very grateful !

**Reply by bezo97** [reply]
I've enabled all optimizations I could. Sadly the trimming idea is not possible due to how the IFS algorithm works, we need to calculate everything even if it's not on the image. One tip I can give is to try to avoid having very dense areas in the image - because writing to the histogram is currently the bottleneck and one dense bucket can stall many threads.. I'm thinking of ways to solve this but not sure yet 🤔 maybe multiple histogram buffers or so in the future.
Also I've found that decreasing syntropy and warmup counter can reduce noise, but these also heavily change how the fractal looks. You can try to set these at the start of exploration and see if the fractals look good

**Reply by MaxStarovoyt** [reply]
Yes, I'll try your suggestion. Thank you!

**Reply by MaxStarovoyt** [reply]
it may be possible to add additional processing CPU resources?
 to make the GPU and CPU work together

**Reply by bezo97** [reply]
That would require a rewrite of the whole engine (in opencl maybe, instead of opengl) - not on my list :D
